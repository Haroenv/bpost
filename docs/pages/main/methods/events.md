---
title: Getting package events
permalink: events
sidebar: main
---
## `#droppedOffBySender`

### Description
`bpost.droppedOffBySender()` will return information when the package is dropped off by the sender at sender's source postal point.
It requires one passed parameter `id` and returns the data with the `key: 'received.domestic-corner',` value in the API response.

### Spec

|Parameters|Return type|
|---|---|
|`(id: String)`|`Promise<Object>`|

### Usage
```js
import { droppedOffBySender } from 'bpost';

droppedOffBySender(`323267009691651472`).then(res => res);
```

### Output
```js
{
  date: "2016-11-09",
  irregularity: false,
  key: "received.domestic-corner",
  location: {
    en: "PP PRESS SHOP WAASLAND",
    fr: "PP PRESS SHOP WAASLAND",
    nl: "PP PRESS SHOP WAASLAND",
  },
  time: "18:04",
}
```

## `#availableForPickup`

### Description
`bpost.availableForPickup()` will return if a package is available for pickup by the recipient.
It requires one passed parameter `id` and returns a true or false if the object with `knownProcessStep: 'AVAILABLE_IN_POST_POINT_INTERNATIONAL'` has `obj.status === 'active'` in the API response.

### Spec

|Parameters|Return type|
|---|---|
|`(id: String)`|`Promise<Boolean>`|

### Usage
```js
import { availableForPickup } from 'bpost';

availableForPickup(`323267009691651472`).then(res => res);
```

### Output
```js
true // when processOverview.processSteps.status === 'active'
false // when processOverview.processSteps.status !== 'active'
```
