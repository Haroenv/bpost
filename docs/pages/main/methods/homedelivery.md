---
title: Home Delivery
permalink: homedelivery
sidebar: main
---
## `#availableFrom`

### Description
`bpost.availableFrom()` will return the earliest timestamp that a package is availble for pickup.
It requires one passed parameter `id` returns the data with the `earliestAvailableTime: {…}` value in the API response.

{% include warning.html
  content='This method is only available if the recipient was absent on first delivery attempt.<br> It will return `undefined` if the recipient was present at first delivery attempt.'
%}

### Spec

|Parameters|Return type|
|---|---|
|`(id: String)`|`Promise<Object>`|

### Usage
```js
import { availableFrom } from 'bpost';

availableFrom(`323267009691651472`).then(res => res);
```

### Output
```js
{
  time: "11:00",
  day: "2016-11-23"
}
```

## `#availableTo`

### Description
`bpost.availableTo()` will return the latest timestamp that a package can be picked up.
It requires one passed parameter `id` returns the data with the `latestAvailableTime: {…}` value in the API response.

{% include warning.html
  content='This method is only available if the recipient was absent on first delivery attempt.<br> It will return `undefined` if the recipient was present at first delivery attempt.'
%}

### Spec

|Parameters|Return type|
|---|---|
|`(id: String)`|`Promise<Object>`|

### Usage
```js
import { availableTo } from 'bpost';

availableTo(`323267009691651472`).then(res => res);
```

### Output
```js
{
  time: "22:00",
  day: "2016-12-07"
}
```

## `#availableFromTo`

### Description
`bpost.availableFromTo()` will return a formatted string containing the period (in days) that a package is available for pickup after absence.
It requires one passed parameter `id` returns a combined data for the values of `earliestAvailableTime.day` and `latestAvailableTime.day` in the API response.

{% include warning.html
  content='This method is only available if the recipient was absent on first delivery attempt.<br> It will return `undefined` if the recipient was present at first delivery attempt.'
%}

### Spec

|Parameters|Return type|
|---|---|
|`(id: String)`|`Promise<String>`|

### Usage
```js
import { availableFromTo } from 'bpost';

availableFromTo(`323267009691651472`).then(res => res);
```

### Output
```js
'Available from 2016-11-23 to 2016-12-07'
```

## `#attemptedDelivery`

### Description
`bpost.attemptedDelivery()` will return the timestamp for the first delivery attempt in which recipient was absent.
It requires one passed parameter `id` returns the data with the `attemptedDeliveryTime: {…}` value in the API response.

### Spec

|Parameters|Return type|
|---|---|
|`(id: String)`|`Promise<Object>`|

### Usage
```js
import { attemptedDelivery } from 'bpost';

attemptedDelivery(`323267009691651472`).then(res => res);
```

### Output
```js
{
  time: "12:28",
  day: "2016-11-22"
}
```

## `#actualDelivery`

### Description
`bpost.actualDelivery()` will return the timestamp for when the package was picked up at the postal point after absence.
It requires one passed parameter `id` returns the data with the `actualDeliveryTime: {…}` value in the API response.

{% include warning.html
  content='This method is only available if a package is picked up after first delivery attempt.<br> It will return `undefined` if the recipient was present at first delivery attempt or if the package is yet to be picked up at the postal point.'
%}

### Spec

|Parameters|Return type|
|---|---|
|`(id: String)`|`Promise<Object>`|

### Usage
```js
import { actualDelivery } from 'bpost';

actualDelivery(`323267009691651472`).then(res => res);
```

### Output
```js
{
  time: "13:37",
  day: "2016-11-23"
}
```
