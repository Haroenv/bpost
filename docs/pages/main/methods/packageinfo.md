---
title: Package Info
permalink: packageinfo
sidebar: main
---
#### packageInfo.js
__bpost#getPackageWeight()__
> Number: returns the package weight in grams

__bpost#deliveryMethod()__
> String: returns the requested delivery method
(e.g `"POSTAL_POINT"`)

__bpost#identifier()__
> Number: returns the internal id for package
_❗ this is different from the itemIdentifier param passed to the methods as first argument_

__bpost#hasRetourOptions()__
> Boolean: returns true when package is eligible for retour

__bpost#receiver(formatted: Boolean)__
> Object: returns data for the receiver
> String: returns a formatted string with receiver data
(eg. `"THIBAULT MAEKELBERGH, 8000 BRUGGE, BE"`)
