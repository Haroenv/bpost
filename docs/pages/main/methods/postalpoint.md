---
title: Postal Point
permalink: postalpoint
sidebar: main
---
## `#sourcePostalPoint`

### Description
`bpost.sourcePostalPoint()` will return the name of the postal point where the sender drops off the package.
It requires one passed parameter `id` returns the data for the `location.en` key proceeding the `"key":"received.domestic-corner"` pair.

It supports [multi-language returns](/multilanguage), either [constructed with a session](/constructing) or passed directly as a parameter.

### Spec

|Parameters|Return type|
|---|---|
|`(id: String)`|`Promise<String>`|
|`(id: String, ?en: String)`|`Promise<String>`|

### Usage
```js
import { sourcePostalPoint } from 'bpost';

sourcePostalPoint(`323267009691651472`).then(res => res);
sourcePostalPoint(`323267009691651472`, 'nl').then(res => res);
```

### Output
```js
'PP PRESS SHOP WAASLAND'
```

## `#destinationPostalPoint`

### Description
`bpost.destinationPostalPoint()` will return the name of the postal point chosen as destination.
It requires one passed parameter `id` returns the data with the `events.deliveryPoint.name.en: …` value in the API response.

It supports [multi-language returns](/multilanguage), either [constructed with a session](/constructing) or passed directly as a parameter.

### Spec

|Parameters|Return type|
|---|---|
|`(id: String)`|`Promise<String>`|
|`(id: String, ?lang: String)`|`Promise<String>`|

### Usage
```js
import { destinationPostalPoint } from 'bpost';

destinationPostalPoint(`323267009691651472`).then(res => res);
destinationPostalPoint(`323267009691651472`, `nl`).then(res => res);
```

### Output
```js
'POSTPUNT OCTA+ BRUGGE'
```

## `#destinationAsAddress`

### Description
`bpost.destinationAsAddress()` will return a formatted string containing the address of the destination postal point.
It requires one passed parameter `id` returns the data with the `deliveryPoint.en: …` value in the API response.

It supports [multi-language returns](/multilanguage), either [constructed with a session](/constructing) or passed directly as a parameter.

### Spec

|Parameters|Return type|
|---|---|
|`(id: String)`|`Promise<String>`|
|`(id: String, ?lang: String)`|`Promise<String>`|

### Usage
```js
import { destinationAsAddress } from 'bpost';

destinationAsAddress(`323267009691651472`).then(res => res);
destinationAsAddress(`323267009691651472`, nl).then(res => res);
```

### Output
```js
'Pathoekeweg 74, 8000 Brugge'
```

## `#openingHours`

### Description
`bpost.openingHours()` will return the opening and closing hours the destination postal point on a given day.
It requires two passed parameters `id` and `day` and returns the data with the `deliveryPoint.openingSchedules: {…}` value in the API response.


### Spec

|Parameters|Return type|
|---|---|
|`(id: String, day: String)`|`Promise<Object>`|

### Usage
```js
import { openingHours } from 'bpost';

openingHours(`323267009691651472`, 'monday').then(res => res);
```

### Output
```js
{
  open: "07:00",
  close: "19:00",
}
```

## `#destinationCoordinate`

### Description
`bpost.destinationCoordinate()` will return a 2D lat long coordinate for the destination postal point.
It requires one passed parameter `id` returns the data with the `deliveryPoint.latitude: …` and `deliveryPoint.longitude: …` values in the API response, formatted as an object.

### Spec

|Parameters|Return type|
|---|---|
|`(id: String)`|`Promise<Object>`|

### Usage
```js
import { destinationCoordinate } from 'bpost';

destinationCoordinate(`323267009691651472`).then(res => res);
```

### Output
```js
{
  longitude: 3.21387,
  latitude: 51.23154,
}
```
