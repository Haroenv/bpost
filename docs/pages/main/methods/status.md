---
title: Getting active status
permalink: status
sidebar: main
---
## `#getStatus`

### Description
`bpost.getStatus()` will search for the currently active step in the delivery process of the package.
It requires one passed parameter `id` and returns the data with the `status: 'active'` value in the API response.

### Spec

|Parameters|Return type|
|---|---|
|`(id: String)`|`Promise<Object>`|

### Usage
```js
import { getStatus } from 'bpost';

getStatus(`323267009691651472`).then(res => res);
```

### Output
```js
{
  knownProcessStep: "PICKED_UP_IN_POST_POINT_INTERNATIONAL",
  label: {
    detail: "inPostPoint",
    main: "pickedUp",
  },
  name: "delivered",
  status: "active",
}
```
