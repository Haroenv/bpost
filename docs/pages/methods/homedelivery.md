---
layout: method
---
#### homeDeliveryOptions.js
__bpost#availableFrom()__
> Object: returns the earliest time and day that a package can be picked up after absence

__bpost#availableTo()__
> Object: returns the latest time and day that a package can be picked up after absence

__bpost#availableFromTo()__
> String: returns a formatted string stating the period (in days) a package can be picked up after absence
(e.g `"Available from 2016-11-23 to 2016-12-07"`)

__bpost#attemptedDelivery()__
> Object: returns the time and day that delivery was attempted but receiver was absent

__bpost#actualDelivery()__
> Object: returns the time and day when a package was picked up from the postal point, receiving the absently delivered package
