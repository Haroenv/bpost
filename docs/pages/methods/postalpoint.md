---
layout: method
---
#### postalPointOptions.js
__bpost#sourcePostalPoint()__
> String: returns the English name for the postal point where your package is delivered by the sender

__bpost#destinationPostalPoint()__
> String: returns the English name for the postal point where your package is delivered for the recipient

__bpost#destinationAsAddress()__
> String: returns a formatted string with the complete destination address
(eg. `"Pathoekeweg 73, 8000 Brugge"`)

__bpost#openingHours(day: String)__
> Object: returns opening and closing times on the given day
(eg. `{ open: "07:00", close: "19:00" }`)

__bpost#destinationCoordinate()__
> Object: returns a 2D lat long coordinate of the destination
(eg. `{ "longitude": 3.21387, "latitude": 51.23154, }`)
