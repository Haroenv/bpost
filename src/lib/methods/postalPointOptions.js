import find from 'lodash.find';
import composedFetch from '../composedFetch.js';

export const sourcePostalPoint = (id, lang = `en`) => composedFetch(id)
  .then(data => {
    const result = find(data.events, { key: `received.domestic-corner` });
    return result.location[lang];
  });

export const destinationPostalPoint = (id, lang = `en`) => composedFetch(id)
  .then(data => data.deliveryPoint.name[lang]);

export const destinationAsAddress = (id, lang = `en`) => composedFetch(id)
  .then(data => {
    const obj = data.deliveryPoint;
    return `${obj.street[lang]} ${obj.streetNumber[lang]}, ${obj.postcode[lang]} ${obj.municipality[lang]}`; // eslint-disable-line max-len
  });

export const openingHours = (id, day) => {
  if (!day) throw new TypeError(`Please specify a day`);

  return composedFetch(id)
    .then(data => {
      const obj = find(
        data.deliveryPoint.openingSchedules,
        { dayOfTheWeek: day.toUpperCase() }
      );

      return obj.openingHours[0];
    });
};

export const destinationCoordinate = id => composedFetch(id)
  .then(data => {
    return {
      latitude: data.deliveryPoint.latitude,
      longitude: data.deliveryPoint.longitude,
    };
  });
