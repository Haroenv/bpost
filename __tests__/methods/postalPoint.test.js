import * as bpost from '../../src/lib/methods/postalPointOptions';

describe(`Method: sourcePostalPoint`, () => {
  it(`should return the source postal point name`, async () => {
    const request = await bpost.sourcePostalPoint(global.BPOST_ID_POSTAL);
    expect(request).toEqual(`PP PRESS SHOP WAASLAND`);
  });
});

describe(`Method: destinationPostalPoint`, () => {
  it(`should return the destination postal point name`, async () => {
    const request = await bpost.destinationPostalPoint(global.BPOST_ID_POSTAL);
    expect(request).toEqual(`POSTPUNT OCTA+ BRUGGE`);
  });
});

describe(`Method: destinationAsAddress`, () => {
  it(`should return the destination postal point formatted as an address`, async () => {
    const request = await bpost.destinationAsAddress(global.BPOST_ID_POSTAL);
    expect(request).toEqual(`PATHOEKEWEG 74-78, 8000 BRUGGE`);
  });
});

describe(`Method: openingHours`, () => {
  it(`should fail if no day is passed`, async () => {
    try {
      await bpost.openingHours(global.BPOST_ID_POSTAL);
    } catch (e) {
      expect(e).toThrow();
    }
  });

  it(`should return the opening hours`, async () => {
    const request = await bpost.openingHours(global.BPOST_ID_POSTAL, `monday`);
    expect(request).toMatchSnapshot(`openingHours`);
  });
});

describe(`Method: destinationCoordinate`, () => {
  it(`should return destination as 2D coordinate`, async () => {
    const request = await bpost.destinationCoordinate(global.BPOST_ID_POSTAL);
    expect(request).toMatchSnapshot(`destinationCoordinate`);
  });
});
