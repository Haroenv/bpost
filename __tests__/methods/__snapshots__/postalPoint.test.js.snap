exports[`Method: destinationCoordinate should return destination as 2D coordinate 1`] = `
Object {
  "latitude": 51.23154,
  "longitude": 3.21387,
}
`;

exports[`Method: openingHours should return the opening hours 1`] = `
Object {
  "close": "19:00",
  "open": "07:00",
}
`;
